report-service/src/main/kotlin/ru/morkovka/report/service/impl/ReleaseServiceImpl.kt

```diff
@@ -48,36 +50,46 @@ class ReleaseServiceImpl(
                             || comment.startsWith(taskCommentDeployInstructionsStart)
                 }.collect(Collectors.toList())
             comments[task.key] = task.comments
-        }
+        } 
 
         logger.info("getTasksTestingAndDeployInfoByJiraRelease [jiraFixVersion = $jiraFixVersion]: jira search completed")
 
-        // TODO change default comparator to custom one. "DM-1" < "DM-2" < ... < "DM-10" < "DM-11" < ...
-        customSort(comments) //Custom comparator added
-        return comments
+        return customSort(comments) //Custom comparator added
     }
 
-    private fun customSort(map: MutableMap<String, MutableList<String>>) {
-        val comparator =
-            Comparator { o1: Map.Entry<String, List<String>>, o2: Map.Entry<String, List<String>> ->
-                val one = getNumber(o1.key)
-                val two = getNumber(o2.key)
-                one.compareTo(two)
-            }
-        val entries: List<Map.Entry<String, MutableList<String>>> =
-            ArrayList<Map.Entry<String, MutableList<String>>>(map.entries)
-        entries.sortedWith(comparator)
-        map.clear()
-        for ((key, value) in entries) {
-            map[key] = value
-        }
+    fun customSort(map: MutableMap<String, MutableList<String>>):
+            MutableMap<String, MutableList<String>> {
+//        val comparator =
+//            Comparator { o1: Map.Entry<String, List<String>>, o2: Map.Entry<String, List<String>> ->
+//                val one = getNumber(o1.key)
+//                val two = getNumber(o2.key)
+//                one.compareTo(two)
+//            }
+//        val entries: List<Map.Entry<String, MutableList<String>>> =
+//            ArrayList<Map.Entry<String, MutableList<String>>>(map.entries)
+//        entries.sortedWith(comparator)
+//        map.clear()
+//        for ((key, value) in entries) {
+//            map[key] = value
+//        }
+        return map.toSortedMap(compareBy<String> { getProjectName(it) }.thenBy { getNumber(it) })
     }
```

Comment:
Когда используем систему контроля версий (гит), нужно приучать себя удалять старый код, не пугаясь. Если он вдруг понадобится, то всегда можно посмотреть по истории изменений.

Когда лишние строчки с комментариями удалятся, метод будет состоять из одного метода `map.toSortedMap(...)`. Котлин позволяет еще немного упростить код:
```suggestion
    fun customSort(map: MutableMap<String, MutableList<String>>) =
        map.toSortedMap(compareBy<String> { getProjectName(it) }.thenBy { getNumber(it) })
```